server/src/main/java/io/crate/analyze/relations/select/SelectAnalyzer.java

```diff
@@ -49,13 +53,50 @@ public static SelectAnalysis analyzeSelectItems(List<SelectItem> selectItems,
                                                     Map<RelationName, AnalyzedRelation> sources,
                                                     ExpressionAnalyzer expressionAnalyzer,
                                                     ExpressionAnalysisContext expressionAnalysisContext) {
+        selectItems = adjustSelectItemsFromAliasedTableFunctions(selectItems, sources);
         SelectAnalysis selectAnalysis = new SelectAnalysis(
             selectItems.size(), sources, expressionAnalyzer, expressionAnalysisContext);
         selectItems.forEach(x -> x.accept(INSTANCE, selectAnalysis));
         SelectSymbolValidator.validate(selectAnalysis.outputSymbols());
         return selectAnalysis;
     }
 
+    private static List<SelectItem> adjustSelectItemsFromAliasedTableFunctions(List<SelectItem> selectItems,
+                                                                               Map<RelationName, AnalyzedRelation> sources) {
+        if (sources.values().stream()
+            .filter(x -> x instanceof AliasedAnalyzedRelation)
+            .noneMatch(y -> ((AliasedAnalyzedRelation) y).relation() instanceof TableFunctionRelation))
+            return selectItems;
+
+        var relNames = sources.keySet().stream().map(RelationName::name).collect(Collectors.toList());
+
+        return selectItems.stream()
+            .map(
+                item -> {
+                    if (item instanceof SingleColumn) {
+                        var col = (SingleColumn) item;
+                        String alias = col.toString().substring(1, col.toString().length() - 1);
+                        if (col.getAlias() == null && relNames.contains(alias)) {
+                            return new SingleColumn(
```

Comment:
instead of `new SingleColumn` probably could `new AliasSymbol` within `SingleColumn` visitor method